ui/main.qml

```diff
@@ -222,42 +252,31 @@ StatusWindow {
         Theme.changeFontSize(localAccountSensitiveSettings.fontSize)
 
         d.runMockedKeycardControllerWindow()
-
-        startupOnboardingLoader.active = false
     }
 
-    //! Workaround for custom QQuickWindow
     Connections {
         target: applicationWindow
         function onClosing(close) {
-            if (Qt.platform.os === Constants.mac) {
-                loader.sourceComponent = undefined
-                close.accepted = true
-            } else {
-                if (loader.sourceComponent != app) {
-                    Qt.quit();
-                } else if (loader.sourceComponent == app) {
-                    if (localAccountSensitiveSettings.quitOnClose) {
-                        Qt.quit();
-                    } else {
-                        close.accepted = false
-                        applicationWindow.visible = false;
-                    }
-                }
+            // In case not logged in or loading, quit app
+            if (loader.sourceComponent != app) {
+                Qt.quit()
             }
-        }
-    }
-
-    // On MacOS, explicitely restore the window on activating
-    Connections {
-        target: Qt.application
-        enabled: Qt.platform.os === Constants.mac
-        function onStateChanged() {
-            if (Qt.application.state === d.previousApplicationState
-                && Qt.application.state === Qt.ApplicationActive) {
-                makeStatusAppActive()
+            // In case user has set to close should quit app
+            else if (localAccountSensitiveSettings.quitOnClose) {
+                Qt.quit()
```

Comment:
I bet this won't quit the app at all, just close the window.

Reply:
See this shortcut:
```qml
    Action {
        shortcut: StandardKey.Quit
        onTriggered: {
            Qt.exit(0)
        }
    }
```
Under Qt6, if you really want to quit the app, you have to call `exit()` as the `quit()` is blocked

Reply:
<img width="1967" height="466" alt="Image" src="https://github.com/user-attachments/assets/e822cc2a-6517-4b9e-948f-4d48ca8b8129" />

Reply:
Alright, tested, works :shrug: Nice one!